### PR TITLE
Pass each target feature flag separately

### LINKED ISSUE #5017: zig cc: build error when targeting `wasm32-freestanding`
Supplying `-target wasm32-freestanding` to either `zig cc` or `zig c++` results in an error like the following:

```
error: option
      '-atomics,-bulk-memory,-exception-handling,-multivalue,-mutable-globals,-nontrapping-fptoint,-sign-ext,-simd128,-tail-call,-unimplemented-simd128'
      cannot be specified with '-target-feature'
```

clang doesn't seem to like the comma separated list. passing each value separately seems to fix the issue, i.e.

```
-target-feature -atomics -target-feature -bulk-memory -target-feature -exception-handling -target-feature -multivalue -target-feature -mutable-globals -target-feature -nontrapping-fptoint -target-feature -sign-ext -target-feature -simd128 -target-feature -tail-call -target-feature -unimplemented-simd128
```
vs the generated:
```
-target-feature -atomics,-bulk-memory,-exception-handling,-multivalue,-mutable-globals,-nontrapping-fptoint,-sign-ext,-simd128,-tail-call,-unimplemented-simd128
```

In the cc1 TableGen [file](https://github.com/llvm/llvm-project/blob/master/clang/include/clang/Driver/CC1Options.td#L23) `target-feature` is defined as `Separate` (An option which is followed by its value), which seems consistent with the error from the wasm example.

However, the comma separated list does appear to work for all other targets I have tried. So, I have no idea what is going on.

I am using the macOS binary from the website, corresponding to commit 8e9e126d41cbca619a2a4658fb38d10120388e8f

### PR DESCRIPTION
**Pull request by georgeroman**
Fixes #5017

### CODE REVIEW

File: src/codegen.cpp [hunk 1]
```diff
@@ -9355,10 +9355,16 @@ void add_cc_args(CodeGen *g, ZigList<const char *> &args, const char *out_dep_pa
                 args.append(g->zig_target->llvm_cpu_name);
             }
             if (g->zig_target->llvm_cpu_features != nullptr) {
-                args.append("-Xclang");
-                args.append("-target-feature");
-                args.append("-Xclang");
-                args.append(g->zig_target->llvm_cpu_features);
+                // https://github.com/ziglang/zig/issues/5017
+                char *flags = buf_ptr(buf_create_from_str(g->zig_target->llvm_cpu_features));
+                char *flag = strtok(flags, ",");
```
Comment (gereeter) [inline src/codegen.cpp:9360]:
```suggestion
                char *saveptr;
                char *flag = strtok_r(flags, ",", &saveptr);
```
Using the non-threadsafe global-modifying `strtok` is a bit of a footgun. `strtok_r` is safer and, IMO, easier to understand.

Comment (LemonBoy) [reply src/codegen.cpp:9360]: There's also `SplitIterator` in util.cpp

Comment (georgeroman) [reply src/codegen.cpp]: Thanks for the suggestions, I've decided to use `strtok_r` in the end.

Comment (georgeroman) [reply src/codegen.cpp]: It looks like `strtok_r` is POSIX only, I've switched to `SplitIterator`.

**Pull request merged**